tests/framework/test_time.py

```diff
@@ -114,16 +104,66 @@ def test_unequal_steps(SimulationContext, base_config, components):
     # Do a step just to update the next_event_time
     sim.step()
     # Check that next step, we will still update all
-    assert (
-        len(
-            sim._clock.get_active_population(
-                sim.get_population().index,
-                sim._clock.event_time,
-            )
-        )
-        == pop_size
-    )
-    # Check that we actually include 7 in events
+    assert len(active_simulants()) == pop_size
+    # Check that we actually include simulant of index 7 in events
     sim.step()
     for index in listener.event_indexes.values():
         assert 7 in index
+
+
+def test_step_pipeline(SimulationContext, base_config, components):
+    sim = SimulationContext(base_config, components)
+    sim.setup()
+    sim.initialize_simulants()
+    pop_size = len(sim.get_population())
+    active_simulants = lambda: sim._clock.get_active_population(
+        sim.get_population().index,
+        sim._clock.event_time,
+    )
+
+    pipeline = lambda: sim._values.get_value("simulant_step_size")(
+        sim._population.get_population(True).index
+    )
+
+    column = lambda: sim._population._population.step_size
+
+    assert np.all(pipeline() == column())
+    assert len(active_simulants()) == pop_size
```

Comment:
Do we need these `len` asserts? Are they adding any value not present in the above test?

Reply:
No, i suppose not